src/components/FirebaseFunctionsPageTable.tsx

```diff
@@ -0,0 +1,205 @@
+import { FunctionData } from '../types';
+import moment from 'moment';
+import { useSettings } from '../helpers/ContextProvider';
+
+const getElapsedTime = (start: string) => {
+  return moment(start).fromNow();
+};
+
+const columnDefinitions: TableColumn<FunctionData>[] = [
+  {
+    title: 'Name',
+    field: 'name',
+    render: (row: Partial<FunctionData>) => {
+      return (
+        <Box fontWeight="fontWeightBold">
+          <Link
+            target="_blank"
+            href={`https://console.cloud.google.com/functions/details/${row.region}/${row.name}?project=${row.project}`}
+          >
+            {row.name}
+          </Link>
+        </Box>
+      );
+    },
+  },
+  {
+    title: 'Status',
+    field: 'status',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => (
+      <Typography variant="body2" noWrap>
+        {row.status!}
+      </Typography>
+    ),
+  },
+  {
+    title: 'Region',
+    field: 'region',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => (
+      <Typography variant="body2" noWrap>
+        {row.region!}
+      </Typography>
+    ),
+  },
+  {
+    title: 'Last modified',
+    field: 'updateTime',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => (
+      <Typography variant="body2" noWrap>
+        {getElapsedTime(row.updateTime!)}
+      </Typography>
+    ),
+  },
+  {
+    title: 'Runtime',
+    field: 'runtime',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => (
+      <Typography variant="body2" noWrap>
+        {row.runtime}
+      </Typography>
+    ),
+  },
+  {
+    title: 'Memory',
+    field: 'availableMemoryMb',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => (
+      <Typography variant="body2" noWrap>
+        {row.availableMemoryMb} MB
+      </Typography>
+    ),
+  },
+  {
+    title: 'Logs',
+    field: '',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => {
+      return (
+        <Link
+          href={`https://console.cloud.google.com/logs/viewer?project=${row.project}&resource=cloud_function%2Ffunction_name%2F${row.name}%2Fregion%2F${row.region}`}
+          target="_blank"
+        >
+          view logs
+        </Link>
+      );
+    },
+  },
+];
+
+export const FirebaseFunctionsPageTable: React.FC = () => {
+  const [page, setPage] = useState(0);
+  const [pageSize, setPageSize] = useState(5);
+  const [filteredRows, setFilteredRows] = useState<FunctionData[]>([]);
+  const [settings] = useSettings();
+  const [tableProps] = useFirebaseFunctions({
+    project: settings.project,
+    authMethod: settings.authMethod,
+    apiKey: settings.apiKey,
+  });
+
+  useEffect(() => {
+    tableProps.retry();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [settings.authMethod, settings.project]);
+
+  useEffect(() => {
+    setFilteredRows(
+      tableProps.functionsData?.slice(page * pageSize, (page + 1) * pageSize) ??
```

Comment:
Does/can the table library not handle paging internally ?

Reply:
It turned out that all functionality we want is default, so yes :P